### PR TITLE
Add 404 route for all pages that do not exist

### PR DESCRIPTION
**Pull request by benjaminloeffel**
On application start the terminal logs the message: `webpack output is served from /GrowersBrains`.

Create React App assumes the application is hosted at the serving web server's root or a subpath as specified in `package.json` `"homepage": "https://nataliepina.github.io/GrowersBrains",`.

`homepage` in the `package.json` is a requirement for `gh-pages` and we cannot work around it by [serving the same build from different paths](https://create-react-app.dev/docs/deployment/#serving-the-same-build-from-different-paths)

There is a related issue on [StackOverflow](https://stackoverflow.com/questions/43011207/using-homepage-in-package-json-without-messing-up-paths-for-localhost).

The simplest solution appears to be creating an `.env` file with `PUBLIC_URL=/` in the local development environment.

What do you think of this solution @nataliepina @Gajendra-NL @vyasriday ?

### CODE REVIEW

**Comment by vyasriday** [comment]
Please resolve the Conflicts.

File: src/App.js [hunk 1]
```diff
@@ -19,29 +14,28 @@ import Error from './screens/Error';
...
-	{ path: '/products', component: Products },
-	{ path: '/dailygrow', component: DailyGrow },
-	{ path: '/consulting', component: Consulting },
-	{ path: '/greenhouse', component: GreenHouse },
+  { path: '/dashboard', component: Dashboard, exact: true },
+  { path: '/products', component: Products },
+  { path: '/dailygrow', component: DailyGrow },
+  { path: '/consulting', component: Consulting },
+  { path: '/greenhouse', component: GreenHouse }
 ];
```
Comment (nataliepina) [inline src/App.js:23]: Something is still broken. Essentially the app is routing to localhost:3000/GrowersBrains which it must think does not exist. Would you be able to troubleshoot this? I believe it should just go to localhost:3000 and load the landing page but it is not doing that and it is trying to start the application routed to /GrowersBrains.

Comment (vyasriday) [reply src/App.js:23]: I think that has got something to do with the `homepage` set in `package.json`.

Comment (vyasriday) [reply src/App.js:23]: I was able to fix it using https://create-react-app.dev/docs/deployment/#serving-the-same-build-from-different-paths that @benjaminloeffel provided.

Comment (benjaminloeffel) [reply src/App.js:23]: @nataliepina  

The [previous commit](https://github.com/nataliepina/GrowersBrains/pull/32/commits/3d2587649604662357a5e69ebb66ed645445979a) just cleans up the routing logic. 

The troubleshooting has been done as per my previous comment found below:

> On application start the terminal logs the message: `webpack output is served from /GrowersBrains`.
> 
> Create React App assumes the application is hosted at the serving web server's root or a subpath as specified in `package.json` `"homepage": "https://nataliepina.github.io/GrowersBrains",`.
> 
> `homepage` in the `package.json` is a requirement for `gh-pages` and we cannot work around it by [serving the same build from different paths](https://create-react-app.dev/docs/deployment/#serving-the-same-build-from-different-paths)
> 
> There is a related issue on [StackOverflow](https://stackoverflow.com/questions/43011207/using-homepage-in-package-json-without-messing-up-paths-for-localhost).
> 
> The simplest solution appears to be creating an `.env` file with `PUBLIC_URL=/` in the local development environment.
> 
> What do you think of this solution @nataliepina @Gajendra-NL @vyasriday ?

To resolve this I would have to commit an `.env` file, which we definitely should not.

Comment (vyasriday) [reply src/App.js:23]: @benjaminloeffel I haven't pushed the changes yet. Do you want me to push it along with Conflict Resolve? We don't have to commit an .env file. Those contributing need to have one for themselves.

Comment (benjaminloeffel) [reply src/App.js]: > @benjaminloeffel I haven't pushed the changes yet. Do you want me to push it along with Conflict Resolve? We don't have to commit an .env file. Those contributing need to have one for themselves.

Yes, please you can go ahead and push the changes & conflict resolve. 
Yes, exactly each contributor should set up their own. Perhaps update the readme.md accordingly.

Comment (vyasriday) [reply src/App.js]: Sure! pushed the changes. Please take a pull and see if everything works at your end. Add a .env.local file with following content `PUBLIC_URL= /`

Comment (nataliepina) [reply src/App.js]: That fixed it for me. We'll need to inform the team that they'll need to set that up locally. Could we also add a script to run `npm run start :local`

Comment (vyasriday) [reply src/App.js:23]: Yeah works for me as well. will keep another script in case any issues happen.

**Pull request merged**